### DISCUSSION TITLE
onSubmit unexpected behavior

### DISCUSSION THREAD

**Discussion by assachs** [jquery/jquery#5385] [Q&A] [answered]
Hello,
i have a problem with form submission and found the following. Is this an expected behavior or is this a bug?

There are different ways to define a submit eventhandler of a form:

D1. in the form element: ...form id="f1" onsubmit="alert('test')"..
D2. with plain javascript: document.getElementById('f2').addEventListener('submit', function() {alert('test1');});
D3. with jquery on: $('#f3').on('submit', function() {alert('test2');});

There are also different ways to execute the submission of a form:

E1. With a submit button
the eventhandler is executed regardless how it was defined ( D1, D2, D3)

E2. With javascript submit()
The eventhandler is never called (that is expected according to the documentation)

E3. With javascript requestSubmit()
the eventhandler is executed regardless how it was defined ( D1, D2, D3)

E4. with jquery trigger("submit")
The eventhandler is only executed in case D1 and D3. It is not executed in case D2 (defined with addEventListener).
Is this expected? 

I found this while migration from D1 to D2 and now the events are not executed any more (i am using E4). 

I could migrate from D1 to D3. Or i stay with D2 and also migrate from E4 to E3.

Thanks
Andi

**Comment by dmethvin** [comment, marked as answer]
In general, jQuery only calls event handlers that are attached by jQuery when a method like `.trigger()` is used. Each time you attach an event handler, jQuery adds it to a list of handlers so that it can be called if you use `.trigger()`.  jQuery doesn't know what event handlers may have been attached by other methods. There may be some exception that I'm not thinking about, but that's the general rule.
